plugins/woocommerce/includes/class-wc-session-handler.php

```diff
@@ -496,24 +496,48 @@ private function is_session_expiring() {
 	 * For guests, sessions expire in 48 hours.
 	 */
 	public function set_session_expiration() {
-		$expiring_seconds   = DAY_IN_SECONDS;
-		$expiration_seconds = is_user_logged_in() ? WEEK_IN_SECONDS : 2 * DAY_IN_SECONDS;
-
 		/**
 		 * Filters the session expiration.
 		 *
 		 * @since 5.0.0
 		 * @param int $expiration_seconds The expiration time in seconds.
 		 */
-		$this->_session_expiring = time() + intval( apply_filters( 'wc_session_expiring', $expiring_seconds ) );
-
+		$expiring_seconds = intval( apply_filters( 'wc_session_expiring', DAY_IN_SECONDS ) );
+
+		if ( $expiring_seconds > MONTH_IN_SECONDS - DAY_IN_SECONDS ) {
+			$expiring_seconds = MONTH_IN_SECONDS - DAY_IN_SECONDS;
+			$transient_key    = 'wc_session_handler_warning';
+			if ( false === get_transient( $transient_key ) ) {
+				wc_get_logger()->warning( 'Keeping sessions for longer than 30 days results in perfomance isues, the default is 2 days for guests and 7 days for logged in users, expiry has been capped to 30 days.', array( 'source' => 'wc_session_handler' ) );
```

Comment:
We can make this warning shorter, I think people are aware of the defaults, but they need to know the new limit: 'Session expiring time exceeded maximum limit of 30 days. Capped to prevent performance issues.', We could also rely on $max_expiring_duration / DAY_IN_SECONDS instead of writing 30.